be/src/storage/lake/gc.cpp

```diff
@@ -145,6 +209,28 @@ static Status delete_tablet_metadata(std::string_view root_location, const std::
                 if (locked_tablet_metadata.count(versions[i])) {
                     continue;
                 }
+                // 1, get tablet meta pair
+                auto meta_pair =
+                        get_meta_pair(tablet_mgr, metadata_root_location, tablet_id, versions[i], versions[i + 1]);
+                if (!meta_pair.ok()) {
+                    LOG(ERROR) << "get meta pair failed " << meta_pair.status().to_string()
+                               << " tablet_id: " << tablet_id;
+                    continue;
+                }
+                std::vector<TabletSegmentId> to_del_tsids;
+                std::unordered_set<int64_t> free_meta_files;
+                // 2. calculate delvec cache and meta files to be free.
+                calc_free_delvec_cache(*meta_pair, to_del_tsids);
+                calc_free_meta_files(*meta_pair, free_meta_files);
+                // 3. free delvec cache
+                tablet_mgr->update_mgr()->clear_cached_del_vec(to_del_tsids);
+                // 4. free useless meta files
+                for (const int64_t ver : free_meta_files) {
+                    auto path = join_path(metadata_root_location, tablet_metadata_filename(tablet_id, ver));
+                    LOG(INFO) << "Deleting " << path;
+                    auto st = fs->delete_file(path);
+                    LOG_IF(WARNING, !st.ok() && !st.is_not_found()) << "Fail to delete " << path << ": " << st;
+                }
```

Comment:
These are primary key model-specific logics and should be skipped if the current table is not a primary key model